src/SFA.DAS.FAT.Web.UnitTests/Controllers/CoursesControllerTests/WhenGettingCourses.cs

```diff
@@ -35,5 +38,30 @@ public class WhenGettingCourses
             var actualModel = actualResult.Model as CoursesViewModel;
             Assert.IsNotNull(actualModel);
         }
+
+        [Test, MoqAutoData]
+        public async Task Then_The_Keyword_Is_Added_To_The_Query_And_Returned_To_The_View(
+            CoursesRouteModel routeModel,
+            GetCoursesResult response,
+            [Frozen] Mock<IMediator> mediator)
+        {
+            //Arrange
+            var controller = new CoursesController(mediator.Object);
+            mediator.Setup(x => x.Send(It.IsAny<GetCoursesQuery>(), It.IsAny<CancellationToken>())).ReturnsAsync(response);
+
+            //Act
+            var actual = await controller.Courses(routeModel);
+
+            //Assert
+            Assert.IsNotNull(actual);
+            var actualResult = actual as ViewResult;
+            Assert.IsNotNull(actualResult);
+            var actualModel = actualResult.Model as CoursesViewModel;
+            Assert.IsNotNull(actualModel);
+            actualModel.Keyword.Should().Be(routeModel.Keyword);
+            mediator.Verify(
```

Comment:
Would be tempted to move this up into the arrange instead, so that is more specific. If it fails then assert on the model should make the test fail